File: OLD_SRC_2/util/network/INetworkPacket.java
package com.sleepwalker.sleeplib.util.network;

import net.minecraft.network.PacketBuffer;
import net.minecraftforge.fml.network.NetworkDirection;
import net.minecraftforge.fml.network.NetworkEvent;

import javax.annotation.Nonnull;

public interface INetworkPacket {

    @Nonnull
    NetworkDirection getNetworkDirection();

    void write(@Nonnull PacketBuffer pf);

    void handle(@Nonnull NetworkEvent.Context context);
}

File: src/main/kotlin/com/sleepwalker/sleeplib/util/network/INetworkPacket.kt
package com.sleepwalker.sleeplib.util.network

import net.minecraft.network.PacketBuffer
import net.minecraftforge.fml.network.NetworkDirection
import net.minecraftforge.fml.network.NetworkEvent

interface INetworkPacket {
    val networkDirection: NetworkDirection
    fun write(buffer: PacketBuffer)
    fun handle(context: NetworkEvent.Context)
}